mods/ra/maps/infiltration/infiltration.lua

```diff
@@ -0,0 +1,355 @@
+	end)[1]
+	BridgeTank.Attack(bridge, true, true)
+end
+
+InfiltrateLabFailed = function()
+	Utils.Do(humans, function(player)
+		if player then
+			player.MarkFailedObjective(infiltrateLab)
+		end
+	end)
+end
+
+ChangeOwnerOnAddedToWorld = function(actor, newOwner)
+	Trigger.OnAddedToWorld(actor, function(unloadedActor)
+		unloadedActor.Owner = newOwner
+	end)
+end
+
+InsertSpies = function()
+	Utils.Do(humans, function(player)
+		if player then
+			infiltrateLab = player.AddPrimaryObjective("Get our spy into the laboratory undetected.")
+		end
+	end)
+
+	Trigger.OnKilled(Lab, function()
+		if not player1.IsObjectiveCompleted(infiltrateLab) then
+			InfiltrateLabFailed()
+		end
+	end)
+
+	-- The delay isn't purely cosmetic, but also prevents a System.InvalidOperationException
+	-- "Collection was modified after the enumerator was instantiated." in tick_activities
+	Trigger.OnInfiltrated(Lab, function()
+		Trigger.AfterDelay(DateTime.Seconds(3), LabInfiltrated)
+	end)
+
+	spyActors = { "spy.strong" }
+	if player2 then
+		spyActors = { "spy.strong", "spy.strong" }
+	end
+
+	local entryPath = { SpyReinforcementsEntryPoint.Location, SpyReinforcementsUnloadPoint.Location }
+	local exit = { SpyReinforcementsExitPoint.Location }
+	local reinforcements = Reinforcements.ReinforceWithTransport(allies, TransportType, spyActors, entryPath, exit)
+
+	local transport = reinforcements[1]
+	Camera.Position = transport.CenterPosition
+
+	spies = reinforcements[2]
+	Trigger.OnAnyKilled(spies, InfiltrateLabFailed)
+
+	ChangeOwnerOnAddedToWorld(spies[1], player1)
+
+	if player2 then
+		ChangeOwnerOnAddedToWorld(spies[2], player2)
+	end
+end
+
+IdleHunt = function(unit)
+	Trigger.OnIdle(unit, unit.Hunt)
+end
+
+StopHunt = function(unit)
+	Trigger.Clear(unit, "OnIdle")
+end
+
+AttackTown = function()
+	Utils.Do(TownAttackers, IdleHunt)
+
+	Trigger.OnRemovedFromWorld(Hospital, function()
+		Utils.Do(TownAttackers, StopHunt)
+	end)
+end
+
+CapOre = function(player)
+	if player.Resources > player.ResourceCapacity then
+		player.Resources = player.ResourceCapacity * 0.8
+	end
+end
+
+NewPatrol = function(actorType, start, waypoints)
+	local guard = Actor.Create(actorType, true, { Owner = soviets, Location = start })
+	guard.Patrol(waypoints, true, Utils.RandomInteger(50, 75))
+end
+
+SetupPatrols = function()
+	Utils.Do(Patrol1, function(patrol1) NewPatrol(patrol1, PatrolPoints1[1], PatrolPoints1) end)
+	Utils.Do(Patrol2, function(patrol2) NewPatrol(patrol2, PatrolPoints1[3], PatrolPoints1) end)
+	Utils.Do(Patrol2, function(patrol3) NewPatrol(patrol3, PatrolPoints3[1], PatrolPoints3) end)
+	Utils.Do(Patrol2, function(patrol4) NewPatrol(patrol4, PatrolPoints4[1], PatrolPoints4) end)
+
+	if Difficulty == "hard" then
+		Utils.Do(Patrol3, function(patrol5) NewPatrol(patrol5, PatrolPoints2[1], PatrolPoints2) end)
+	end
+
+	local checkpoint = { BaseGuardTruckPos.Location }
+	Trigger.OnEnteredFootprint(checkpoint, function()
+		BaseGuard.ScriptedMove(BaseGuardMovePos.Location)
```

Comment:
You can kill him later in the game (with tanks and other units), as the trigger isn't removed.